README-extensions.rst

```diff
@@ -569,3 +569,117 @@ This adds the subfolder to the node name and the structure above can then be use
       ...
 
 If the subfolder path starts with the underscore character ``_``, then the subfolder path is NOT added to the node name.
+
+
+Git storage type
+----------------
+
+Reclass node and class yaml files can be read from a remote git repository with the yaml_git storage type. Use nodes_uri and
+classes_uri to define the git repos to use for nodes and classes. These can be the same repo.
+
+For salt masters using ssh connections the private and public keys must be readable by the salt daemon, which requires the
+private key NOT be password protected. For stand alone reclass using ssh connections if the privkey and pubkey options
+are not defined then any in memory key (from ssh-add) will be used.
+
+Salt master reclass config example:
+
+.. code-block:: yaml
+
+  storage_type:yaml_git
+  nodes_uri:
+    # branch to use
+    branch: master
+
+    # cache directory (default: ~/.reclass/git/cache)
+    cache_dir: /var/cache/reclass/git
+
+    # lock directory (default: ~/.reclass/git/lock)
+    lock_dir: /var/cache/reclass/lock
```

Comment:
I bit dislike, we do a general lock (file), not the lock per process.
You know, there can be two one from CLI one from Salt. Another tool, etc.

Another thing is where the default cache point's to - IMPO kind of OS TEMP dir would fit better.

Reply:
Having a single git repo cache and single lock file handling both reclass cli and salt use would mean either all reclass users plus the salt daemon having write access to the git cache or having a daemon process to manage updating the git repo cache. Having a separate cache per user and an additional cache for salt use gets around this issue and keeps things straight forward.

The important thing is that the salt master uses a single lock file per git repo regardless of which salt master process is doing the work. It also guarantees that if there are git problems then restarting the salt master will clear any reclass related git locks and that people having problems with the git caches using the reclass  cli with not interfere with the salt master copy of the git repos.

I agree that the default directory choice is a pain. For reclass cli use using a .reclass directory in a users home directory is a good default but for salt use the default should indeed be an os directory the salt master daemon can write to. For the redhat systems I'm running on using /var/cache/reclass would be a good default. I went with the ~/.reclass default for now as this means setting a value only in the salt master config and then things just work for reclass cli users.

Thinking about it, this looks like a new minor feature of allowing different defaults when running the reclass cli or via salt would be useful. But that needs some thought about how to do it.